trestle/core/commands/describe.py

```diff
@@ -0,0 +1,149 @@
+            model_file = pathlib.Path(args.file)
+
+            element = '' if 'element' not in args or args.element is None else args.element.strip("'")
+
+            results = self.describe(model_file, element)
+            return 0 if len(results) > 0 else 1
+
+        logger.warning('No file specified for command describe.')
+        return 1
+
+    @classmethod
+    def _clean_type_string(cls, text: str) -> str:
+        text = text.replace("<class '", '').replace("'>", '')
+        text = text.replace('trestle.oscal.', '')
+        text = text.replace('pydantic.main.', 'stripped.')
+        return text
+
+    @classmethod
+    def _description_text(cls, sub_model: OscalBaseModel) -> str:
+        clip_string = 100
+        if sub_model is None:
+            return 'None'
+        if type(sub_model) is list:
+            n_items = len(sub_model)
+            type_text = 'Unknown' if not n_items else f'{cls._clean_type_string(str(type(sub_model[0])))}'
+            text = f'list of {n_items} items of type {type_text}'
+            return text
+        if type(sub_model) is str:
+            return sub_model if len(sub_model) < clip_string else sub_model[:clip_string] + '[truncated]'
+        if hasattr(sub_model, 'type_'):
+            return cls._clean_type_string(str(sub_model.type_))
+        return cls._clean_type_string(str(type(sub_model)))
+
+    @classmethod
+    def describe(cls, file_path: pathlib.Path, element_path_str: str) -> List[str]:
+        """Describe the contents of the file.
+
+        Args:
+            file_path: pathlib.Path Path for model file to describe.
+            element_path_str: Element path of element in model to describe.  Can be ''.
+
+        Returns:
+            The list of lines of text in the description, or an empty list on failure
+        """
+        # figure out the model type so we can read it
+        try:
+            model_type, _ = fs.get_stripped_contextual_model(file_path)
+            model: OscalBaseModel = model_type.oscal_read(file_path)
+        except TrestleError as e:
+            logger.warning(f'Error loading model {file_path} to describe: {e}')
+            return []
+
+        sub_model = model
+
+        # if an element path was provided, follow the path chain to the desired sub_model
+        if element_path_str:
+            if '*' in element_path_str:
+                logger.warning('Wildcards are not allowed in element paths for describe.')
+                return []
+
+            if '.' not in element_path_str:
+                logger.warning('The element path for describe must either be omitted or contain at least 2 parts.')
+                return []
+
+            element_paths = utils.parse_element_arg(model, element_path_str)
+
+            sub_model_element = Element(model)
+
+            for element_path in element_paths:
+                sub_model = sub_model_element.get_at(element_path, False)
+                sub_model_element = Element(sub_model)
```

Comment:
I'm a bit confused here. it seems like here you are trying to cater to mulitiplle models, however, lines 127-147 seem to be catering to a singular element path.

Reply:
This is indeed confusing - but that's the way element paths work.  The element_path list is a list of links in a chain - so you are following the links to the final model.  There is only one final entity being described.  This is all leveraging the existing code for parse_element_args - which parses the element path string (only one, possibly long one) into a list of element paths - each path being a pair representing a link in the chain.  No wildcards or commas are allowed.  The good thing is that the element path construct does allow this all to happen without much code.

Reply:
Oh - it is true that I don't think I check for commas in the path - so I will add that.